src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -149,6 +149,14 @@ else if (fluid.getType() == Material.WATER) {
 
         return Optional.empty();
     }
+    
+    private boolean isLiquid(Block block) {
+        BlockData data = block.getBlockData();
```

Comment:
This may be a bit performance-intense, so a prior Material check would be very adequate here.

Reply:
> This may be a bit performance-intense, so a prior Material check would be very adequate here.

Done for both fluid in consumeFluid and that.